Mathlib/CategoryTheory/Monoidal/DayConvolution/Closed.lean

```diff
@@ -0,0 +1,282 @@
+there is a unique map `H ⟶ H'` induced by functoriality of ends and functoriality
+of `internalHomDiagramFunctor F`. -/
+def map (ℌ : DayConvolutionInternalHom F G H) {G' : C ⥤ V} {H' : C ⥤ V}
+    (f : G ⟶ G') (ℌ' : DayConvolutionInternalHom F G' H') :
+    H ⟶ H' where
+  app c := Limits.Wedge.IsLimit.lift (ℌ'.isLimitWedge c)
+    (fun j ↦ (ℌ.π c j) ≫
+      (dayConvolutionInternalHomDiagramFunctor
+        F|>.map f|>.app c|>.app (op j)|>.app j))
+    (fun ⦃j j'⦄ φ ↦ by
+      haveI := congrArg (fun t ↦ t.app j') <|
+        dayConvolutionInternalHomDiagramFunctor
+          F|>.map f|>.app c|>.naturality φ.op
+      dsimp at this
+      dsimp
+      rw [Category.assoc, ← (ihom (F.obj j)).map_comp, ← f.naturality,
+        Functor.map_comp, reassoc_of% ℌ.hπ]
+      simp)
+  naturality {c c'} f := by
+    apply Limits.Wedge.IsLimit.hom_ext (ℌ'.isLimitWedge c')
+    intro j
+    dsimp
+    simp only [Category.assoc, map_comp_π]
+    rw [← Limits.Wedge.mk_ι
+        (F := dayConvolutionInternalHomDiagramFunctor F|>.obj _|>.obj c')
+        (H'.obj c') (ℌ'.π c') (ℌ'.hπ c'),
+      ← Limits.Wedge.mk_ι
+        (F := dayConvolutionInternalHomDiagramFunctor F|>.obj _|>.obj c)
+        (H'.obj c) (ℌ'.π c) (ℌ'.hπ c),
+      Limits.Wedge.IsLimit.lift_ι (ℌ'.isLimitWedge c'),
+      Limits.Wedge.IsLimit.lift_ι_assoc (ℌ'.isLimitWedge c) ]
+    simp [← Functor.map_comp]
+
+@[reassoc (attr := simp)]
+lemma map_app_comp_π (ℌ : DayConvolutionInternalHom F G H)
+    {G' : C ⥤ V} {H' : C ⥤ V} (f : G ⟶ G')
+    (ℌ' : DayConvolutionInternalHom F G' H') (c : C) (j : C) :
+    (ℌ.map f ℌ').app c ≫ ℌ'.π c j =
+    ℌ.π c j ≫ (ihom <| F.obj j).map (f.app <| j ⊗ c) := by
+  dsimp [map]
+  rw [← Limits.Wedge.mk_ι
+      (F := dayConvolutionInternalHomDiagramFunctor F|>.obj _|>.obj c)
+      (H'.obj c) (ℌ'.π c) (ℌ'.hπ c),
+    Limits.Wedge.IsLimit.lift_ι (ℌ'.isLimitWedge c)]
+
+section ev
+
+variable [DayConvolution F H] (ℌ : DayConvolutionInternalHom F G H)
+
+/-- Given `ℌ : DayConvolutionInternalHom F H`, if we think of `H.obj G`
+as the internal hom `[F, G]`, then this is the transformation
+corresponding to the component at `G` of the "evaluation" natural morphism
+`F ⊛ [F, _] ⟶ 𝟭`. -/
+def ev_app : F ⊛ H ⟶ G :=
+  DayConvolution.corepresentableBy F H|>.homEquiv.symm <|
+    { app := fun x => MonoidalClosed.uncurry <| ℌ.π x.2 x.1
+      naturality {x y} f := by
+        haveI := congrArg (fun t ↦ F.obj x.1 ◁ t) <| ℌ.hπ x.2 f.1
+        dsimp at this ⊢
+        simp only [whiskerLeft_comp] at this
+        simp only [Category.assoc, MonoidalClosed.uncurry_eq, Functor.id_obj]
+        rw [← whiskerLeft_comp_assoc, map_comp_π]
+        simp [← whisker_exchange_assoc, tensorHom_def,
+          ← ihom.ev_naturality_assoc]
+        rw [reassoc_of% this]
+        simp }
+
+@[reassoc (attr := simp)]
+lemma unit_app_ev_app_app (x y : C) :
+    ((DayConvolution.unit F H).app (x, y) ≫ (ℌ.ev_app).app (x ⊗ y)) =
+    MonoidalClosed.uncurry (ℌ.π y x) := by
+  simp [ev_app]
```

Comment:
Hopefully, doing `simp [ev_app, this]` at the end of the block may fix this nonterminal simp.